packages/asset/contracts/AssetCreate.sol

```diff
@@ -354,5 +356,267 @@ contract AssetCreate is
+            "AssetCreate: Invalid signature"
+        );
+
+        require(mintData.creators.length == mintData.tiers.length, "AssetCreate: 1-Array lengths");
+        require(mintData.amounts.length == mintData.metadataHashes.length, "AssetCreate: 2-Array lengths");
+        require(mintData.metadataHashes.length == mintData.tiers.length, "AssetCreate: 3-Array lengths");
+        require(mintData.tiers.length == mintData.unitPrices.length, "AssetCreate: 4-Array lengths");
+        require(mintData.unitPrices.length == mintData.paymentTokens.length, "AssetCreate: 5-Array lengths");
+        require(mintData.paymentTokens.length == mintData.maxSupplies.length, "AssetCreate: 6-Array lengths");
+
+        uint256[] memory tokenIds = new uint256[](mintData.tiers.length);
+        uint256[] memory tiersToBurn = new uint256[](mintData.tiers.length);
+        for (uint256 i = 0; i < mintData.tiers.length; i++) {
+            uint16 revealed = mintData.tiers[i] == 1 ? 1 : 0;
+            tokenIds[i] = assetContract.getTokenIdByMetadataHash(mintData.metadataHashes[i]);
+            if (tokenIds[i] == 0) {
+                tokenIds[i] = TokenIdUtils.generateTokenId(
+                    mintData.creators[i],
+                    mintData.tiers[i],
+                    ++creatorNonces[mintData.creators[i]],
+                    revealed,
+                    true
+                );
+                require(mintData.amounts[i] <= mintData.maxSupplies[i], "AssetCreate: Max supply exceeded");
+                availableToMint[tokenIds[i]] = mintData.maxSupplies[i] - mintData.amounts[i];
+            } else {
+                require(availableToMint[tokenIds[i]] >= mintData.amounts[i], "AssetCreate: Max supply reached");
+                availableToMint[tokenIds[i]] -= mintData.amounts[i];
+            }
+            distributePayment(
+                mintData.unitPrices[i],
+                mintData.amounts[i],
+                mintData.paymentTokens[i],
+                mintData.creators[i]
+            );
+        }
+
+        catalystContract.burnBatchFrom(_msgSender(), tiersToBurn, mintData.amounts);
+        assetContract.mintBatch(_msgSender(), tokenIds, mintData.amounts, mintData.metadataHashes);
+        emit AssetBatchLazyMinted(
+            _msgSender(),
+            mintData.creators,
+            tokenIds,
+            mintData.tiers,
+            mintData.amounts,
+            mintData.metadataHashes
+        );
+    }
+
+    /// @notice Splits the lazy mint payment between the creator and TSB if there is a fee
+    /// @param unitPrice The price of the asset
+    /// @param amount The amount of copies to mint
+    /// @param paymentToken The payment token
+    /// @param creator The address of the creator
+    function distributePayment(
+        uint256 unitPrice,
+        uint256 amount,
+        address paymentToken,
+        address creator
+    ) internal {
+        uint256 fee = (unitPrice * amount * lazyMintFee) / 10000;
```

Comment:
Rather than calculate the fee every time even when the lazyMintFee is zero, can we simply check if there is a lazyMintFee and then move this fee calc into the `if (fee > 0)` block?

Reply:
I've refactored the function to account for that, looking forward to your feedback.